standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnHandler.java

```diff
@@ -5086,8 +5093,10 @@ public void performTimeOuts() {
       while(true) {
         stmt = dbConn.createStatement();
         String s = " \"TXN_ID\" FROM \"TXNS\" WHERE \"TXN_STATE\" = " + TxnStatus.OPEN +
-            " AND \"TXN_LAST_HEARTBEAT\" <  " + getEpochFn(dbProduct) + "-" + timeout +
-            " AND \"TXN_TYPE\" != " + TxnType.REPL_CREATED.getValue();
+            " AND \"TXN_LAST_HEARTBEAT\" <  " + getEpochFn(dbProduct) + "-" +
+            (replication ? replicationTxnTimeout : timeout) +
+            " AND \"TXN_TYPE\" " +
```

Comment:
TXN_TYPE will always be REPL_CREATED for replication. This check looks incorrect.
It could be something like if txn type is REPL_CREATED, use replicationTxnTimeout else timeout

Reply:
@aasha Thanks for taking a look.

performTimeOuts() calls performTimeOutsInternal twice:
1. times out non-repl txns -> timeout is timeout and TXN_TYPE != REPL_CREATED
2. times out repl txns -> timeout is replicationTxnTimeout and TXN_TYPE=REPL_CREATED

I supposed this could be optimized but I don't see how it is incorrect?

Reply:
@aasha I have updated the PR with a more concise/optimal solution